exo/helpers.py

```diff
@@ -54,11 +54,15 @@ def write_used_port(port, used_ports):
 
   while available_ports:
     port = random.choice(list(available_ports))
-    if DEBUG >= 2: print(f"Trying to find available port {port=}")
+    if DEBUG >= 2: print(f"Trying to bind port {port=} on address {host=}")
     try:
       with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
         s.bind((host, port))
-      write_used_port(port, used_ports)
+      try:
+        write_used_port(port, used_ports)
+      except Exception as e:
+        if DEBUG >= 2: print(f"Unable write to file")
```

Comment:
I'd make this a bit more helpful of an error message, for example "Unable to write to file using the write_used_port function" to make this more debuggable.